### PR TITLE
[doc] Add steps how to access VisibilityOnDemand via curl 

### PR DESCRIPTION
**Pull request by xmolitann**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Docs are missing a guide on how to access VisibilityOnDemand via curl, omitting `kubectl get --raw ....`. This is needed if you want to use it programmatically.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/static/examples/admin/cluster-role-and-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kueue-api-get
```
Comment (PBundyra) [inline site/static/examples/admin/cluster-role-and-binding.yaml:4]: Could you please rename this to better reflect its purpose within the Visibility API?

File: site/static/examples/admin/cluster-role-and-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  - "localqueues/pendingworkloads"
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kueue-api-get
```
Comment (PBundyra) [inline site/static/examples/admin/cluster-role-and-binding.yaml:19]: ditto

File: site/static/examples/admin/cluster-role-and-binding.yaml
Comment (PBundyra) [inline site/static/examples/admin/cluster-role-and-binding.yaml:1]: Could you please rename this file to better reflect its purpose within the Visibility API?

**Review by PBundyra — COMMENTED** [review]
Thanks for the PR! Looks good, just a few nits.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 3]
```diff
@@ -172,7 +187,91 @@ You should get results similar to
...
+
+If you followed steps described in [Accessing Cluster/Local Queue via curl](#accessing-cluster-and-local-queue-via-curl) above,
+you can use curl (and jq) to view pending workloads in ClusterQueue `cluster-queue` using following commands:
+
+```bash
+export CLUSTER_NAME="some_server_name"
+TOKEN=$(kubectl get secret default-token -o jsonpath='{.data.token}' | base64 --decode)
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+
+curl -X GET $APISERVER/apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads --header "Authorization: Bearer $TOKEN | jq"
```
Comment (PBundyra) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:200]:
```suggestion
curl -X GET $APISERVER/apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads --header "Authorization: Bearer $TOKEN" --insecure | jq
```

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 4]
```diff
@@ -259,9 +358,55 @@ To view only 1 pending workloads use, starting from position 1 in LocalQueue run
...
+
+If you followed steps described in Accessing Cluster/Local Queue via curl](#accessing-cluster-and-local-queue-via-curl) above,
+you can use curl (and jq) to view pending workloads in LocalQueue `user-queue` using following commands:
 
+```bash
+export CLUSTER_NAME="some_server_name"
+TOKEN=$(kubectl get secret default-token -o jsonpath='{.data.token}' | base64 --decode)
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+
+curl -X GET $APISERVER/apis/visibility.kueue.x-k8s.io/v1alpha1/namespaces/default/localqueues/user-queue/pendingworkloads --header "Authorization: Bearer $TOKEN | jq"
```
Comment (PBundyra) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:406]:
```suggestion
curl -X GET $APISERVER/apis/visibility.kueue.x-k8s.io/v1alpha1/namespaces/default/localqueues/user-queue/pendingworkloads --header "Authorization: Bearer $TOKEN" --insecure | jq
```

File: site/static/examples/admin/cluster-role-and-binding.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (PBundyra) [inline site/static/examples/admin/cluster-role-and-binding.yaml:22]: I think we're missing namespace here

**Review by PBundyra — CHANGES_REQUESTED** [review]

**Comment by PBundyra** [comment]
Due to the need of using `--insecure` flag to avoid using proxy, a user is exposed to MITM attacks which I believe is undesired. Can we find any other alternative or mitigation of this risk? 

Docs: https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#without-kubectl-proxy

**Comment by mimowo** [comment]
@xmolitann would you like / find time to address the comments and complete the review?

**Comment by mbobrovskyi** [comment]
/close

Due to fixed in #3160

**Pull request closed without merging**